KEYCLOAK-17342 Make the default value of default signature algorithm …

Yang-Xie-OSS opened the pull request:
…show up in the admin console

There is no value showed up in Realm settings -> Token -> Default Signature Algorithm in Keycloak's admin console and I fixed it in this PR. The details can be found in this [JIRA](https://issues.redhat.com/projects/KEYCLOAK/issues/KEYCLOAK-17342?filter=allopenissues).

Yang-Xie-OSS committed:
KEYCLOAK-17342 Make the default value of default signature algorithm show up in the admin console

mposolda commented on server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java:
```diff
@@ -187,6 +188,7 @@ public static void importRealm(KeycloakSession session, RealmRepresentation rep,
         if (rep.getNotBefore() != null) newRealm.setNotBefore(rep.getNotBefore());
 
         if (rep.getDefaultSignatureAlgorithm() != null) newRealm.setDefaultSignatureAlgorithm(rep.getDefaultSignatureAlgorithm());
+        else newRealm.setDefaultSignatureAlgorithm(Algorithm.RS256);
```
I suppose that instead of have the default algorithm hardcoded here, it will be better to use some constant and use it from all the places? Basically when we later decide that default algorithm should be changed to for example RS512, we will need to change it in single place and not on many random places inside codebase (which would mean possible bugs as of course some places will be forgotten).

Also will be good to have a test for this.

mposolda reviewed (changes requested):
@Yang-Xie-OSS Thanks for the PR. See the inline comment for more details

Yang-Xie-OSS commented:
@mposolda
Thanks for the review. I've fixed it already. Could you please have a look?
Due to It can't use DefaultTokenManager's method directly in RepresentationToModel, I moved the constant DefaultTokenManager.DEFAULT_ALGORITHM_NAME to org.keycloak.models.Constants.
Besides, I added a test. Unfortunately, It was lots of error happended in local test. But the added method, testDefaultSignatureAlgorithm() was passed successfully. So I think the errors were not related to my code.

Yang-Xie-OSS commented:
@mposolda
In addtion, the last commit can't enable the master realm to show up its default signature algorithm. So I added a line of code in org.keycloak.services.managers.ApplianceBootstrap in this update.

mposolda commented:
@Yang-Xie-OSS Thanks. Changes looks fine at first look, but I think that the test errors in GH actions are caused by your changes? For example see https://github.com/keycloak/keycloak/pull/7835/checks?check_run_id=2083654177 . Could you please doublecheck if running whole RealmTest in your environment works as expected?

Yang-Xie-OSS commented:
@mposolda
Yes, I confirmed these test errors were all caused by my changes. I should remove the added realm after the test.
Now the problem was solved. Please check it.

mposolda commented:
@Yang-Xie-OSS Thanks, but there are still test failures... Before updating PR, you can send running the whole class by running 
```
mvn clean install -DskipTests=true
cd testsuite/integration-arquillian
mvn clean install -Dtest=RealmTest
```
This should give you an idea if it pass or not.

Yang-Xie-OSS commented:
@mposolda 
Thanks for your review.
I must say there were no error occurred in my environment. So I thought the errors may effected by some other test class except RealmTest. Now I did two things as shown below and the errors are resolved.
1. Took the code which will remove the created realm to a finally block so as to it still works even if an exception happened.
2. I found the online error should caused by org.keycloak.testsuite.admin.AdminSignatureAlgorithmTest as it changed the master realm's default signature algorithm to ES256 but didn't return it back. So I added one line to change the default signature algorithm to the before value in finally block.

Now I confirmed all online checks also have passed. Could you please check?

mposolda reviewed (approved):
@Yang-Xie-OSS Great catch with AdminSignatureAlgorithmTest! Thanks!
